src/dataset.jl

```diff
@@ -255,8 +257,36 @@ function writelayers(
     dataset::AbstractDataset,
     filename::AbstractString;
     driver::Driver = getdriver(dataset),
-    options = [""],
-    layer_options = [""],
+    options = String[],
+    layers = 0:(nlayer(dataset)-1),
+    layer_options = String[],
+    chunksize::Integer = 20000,
+    use_gdal_copy::Bool = true,
+)
+    create(filename; driver = driver, options = options) do target
+        return copylayers!(
```

Comment:
Can we add unit test(s) for `copylayer(s)!` without depending on it being called through `writelayers` in the unit tests?